File: python3/677.py
class MapSum:

    def __init__(self):
        self.pairs = {}

    def insert(self, key: str, val: int) -> None:
        if key in self.pairs:
            self.pairs.pop(key)
        self.pairs.setdefault(key, val)

    def sum(self, prefix: str) -> int:
        sum = 0
        for i in self.pairs:
            if i.startswith(prefix):
                sum += self.pairs.get(i)
        return sum


# Your MapSum object will be instantiated and called as such:
# obj = MapSum()
# obj.insert(key,val)
# param_2 = obj.sum(prefix)

File: javascript/677.js
let pair;

var MapSum = function () {
    pair = new Map();
};

/** 
 * @param {string} key 
 * @param {number} val
 * @return {void}
 */
MapSum.prototype.insert = function (key, val) {
    pair.set(key, val);
};

/** 
 * @param {string} prefix
 * @return {number}
 */
MapSum.prototype.sum = function (prefix) {
    let sum = 0;
    for (let i of pair)
        if (i[0].startsWith(prefix))
            sum += i[1];
    return sum;
};

/** 
 * Your MapSum object will be instantiated and called as such:
 * var obj = new MapSum()
 * obj.insert(key,val)
 * var param_2 = obj.sum(prefix)
 */
